plugins/builds/helper/updateBuild.js

```diff
@@ -0,0 +1,304 @@
+'use strict';
+
+const boom = require('@hapi/boom');
+const hoek = require('@hapi/hoek');
+const merge = require('lodash.mergewith');
+const { getFullStageJobName } = require('../../helper');
+const STAGE_TEARDOWN_PATTERN = /^stage@([\w-]+)(?::teardown)$/;
+const TERMINAL_STATUSES = ['FAILURE', 'ABORTED', 'UNSTABLE', 'COLLAPSED'];
+const FINISHED_STATUSES = ['FAILURE', 'SUCCESS', 'ABORTED', 'UNSTABLE', 'COLLAPSED'];
+
+/**
+ * @typedef {import('screwdriver-models/lib/build')} Build
+ * @typedef {import('screwdriver-models/lib/event')} Event
+ */
+
+/**
+ * Identify whether this build resulted in a previously failed job to become successful.
+ *
+ * @method isFixedBuild
+ * @param  build         Build Object
+ * @param  jobFactory    Job Factory instance
+ */
+async function isFixedBuild(build, jobFactory) {
+    if (build.status !== 'SUCCESS') {
+        return false;
+    }
+
+    const job = await jobFactory.get(build.jobId);
+    const failureBuild = await job.getLatestBuild({ status: 'FAILURE' });
+    const successBuild = await job.getLatestBuild({ status: 'SUCCESS' });
+
+    if ((failureBuild && !successBuild) || failureBuild.id > successBuild.id) {
+        return true;
+    }
+
+    return false;
+}
+
+/**
+ * Stops a frozen build from executing
+ * @method stopFrozenBuild
+ * @param  {Object} build         Build Object
+ * @param  {String} previousStatus    Prevous build status
+ */
+async function stopFrozenBuild(build, previousStatus) {
+    if (previousStatus !== 'FROZEN') {
+        return Promise.resolve();
+    }
+
+    return build.stopFrozen(previousStatus);
+}
+
+/**
+ * Updates execution details for init step
+ * @method stopFrozenBuild
+ * @param  {Object} build       Build Object
+ * @param  {Object} app         Hapi app Object
+ */
+async function updateInitStep(build, app) {
+    const step = await app.stepFactory.get({ buildId: build.id, name: 'sd-setup-init' });
+    // If there is no init step, do nothing
+
+    if (!step) {
+        return null;
+    }
+
+    step.endTime = build.startTime || new Date().toISOString();
+    step.code = 0;
+
+    return step.update();
```

Comment:
```suggestion
    return await step.update();
```

Reply:
Same here we don't to `await` here. It is done by the caller 
https://github.com/screwdriver-cd/screwdriver/pull/3252/files/d12977ba7d6469c2ca79437eb26f62eb580b3c5b#diff-2c3decb1d714adeb8f265dd68e51a120caa8ada38d91b21d9dffc66974f3276dR206